neo/rawio/alphaomegarawio.py

```diff
@@ -0,0 +1,1252 @@
+                                "block_type": block_type,
+                                "data": f.read(length),
+                            }
+                        )
+        if prune_channels:
+            to_remove = []
+            for channel_id, channel in continuous_analog_channels.items():
+                if not channel["positions"]:
+                    to_remove.append(channel_id)
+            for channel_id in to_remove:
+                del continuous_analog_channels[channel_id]
+            to_remove = []
+            for channel_id, channel in segmented_analog_channels.items():
+                if not channel["positions"]:
+                    to_remove.append(channel_id)
+            for channel_id in to_remove:
+                del segmented_analog_channels[channel_id]
+            to_remove = []
+            for channel_id, channel in digital_channels.items():
+                if not channel["samples"]:
+                    to_remove.append(channel_id)
+            for channel_id in to_remove:
+                del digital_channels[channel_id]
+            to_remove = []
+            for port_id, port in ports.items():
+                if not port["samples"]:
+                    to_remove.append(port_id)
+            for port_id in to_remove:
+                del ports[port_id]
+            del to_remove
+        return (
+            metadata,
+            continuous_analog_channels,
+            segmented_analog_channels,
+            digital_channels,
+            channel_type,
+            stream_data_channels,
+            ports,
+            events,
+            unknown_blocks,
+        )
+
+    def _merge_segments(self, factor_period=1.5):
+        """This method merge segments that are the same segment but comes from
+        different files. The AlphaOmega data recording system split files with
+        a configurable maximum time and size but with a global non-configurable
+        maximum size of 1GB.
+        Two segment are merged if they validate the following conditions:
+            - start of next segment is less than `factor_period` (default 1.5)
+            the system's sampling period + end of previous segment. We can't
+            check with exactly 1 sampling period because timings are floats
+            which are not exact period values
+            - The two segment must also have the same record date
+            (YEAR-MONTH-DAY). This could potentially lead to errors if
+            recordings are longer than a day or run over the night
+            - The next segment must have a datetime greater or equal than the
+            previous segment (datetime : YEAR-MONTH-DAY-HOUR-MINUTE-SECOND)
+
+        :param factor_period: how many sample period to consider the next segment
+            is part of the previous one. This should be 1 <= `factor_period` < 2,
+            defaults to 1.5
+        :type factor_period: float
+        """
+        for block in self._blocks:
```

Comment:
Same here To avoid any confusion with neo.Block I would use "datablock"' instead of block.